components/brave_wallet/browser/solana_transaction.cc

```diff
@@ -212,6 +212,43 @@ SolanaTransaction::GetSignedTransactionBytes(
   return transaction_bytes;
 }
 
+// Get serialized and signed transaction given a signature from a hardware
+// wallet. Only supports transactions with one signer.
+absl::optional<std::vector<uint8_t>>
+SolanaTransaction::GetSignedTransactionBytes(
+    const std::vector<uint8_t>& signature_bytes) const {
+  if (signature_bytes.size() != kSolanaSignatureSize)
+    return absl::nullopt;
+
+  // Combine the signature from hardware the message and send
+  std::vector<uint8_t> transaction_bytes;
+
+  absl::optional<std::uint8_t> num_signers = message_.GetNumberOfSigners();
+  if (!num_signers || num_signers != 1) {
+    return absl::nullopt;
+  }
+
+  // 1 signer
+  CompactU16Encode(1, &transaction_bytes);
+
+  // Add signature
+  transaction_bytes.insert(transaction_bytes.end(), signature_bytes.begin(),
+                           signature_bytes.end());
+
+  // Add message
+  auto message_bytes = message_.Serialize(nullptr);
```

Comment:
Actually, I don't think `GetNumberOfSigners` is needed here, it makes us running one extra GetUniqueAccountMetas, and for example we are missing UINT8_MAX check.
We can just move this line to above and get signers from it without changing the bytes order we are pushing.